requirements.txt

```diff
@@ -25,16 +24,18 @@ orjson >= 3.7, < 4.0
 packaging >= 21.3, < 24.3
 pathspec >= 0.8.0
 pendulum >= 2.1.2, < 3.0.0
-pydantic >= 1.10.0, < 2.0.0
+# the version constraints for pydantic are merged with those from fastapi 0.103.2
+pydantic>=1.10.0,< 2.0.0,>=1.7.4,!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0
 python-slugify >= 5.0, < 9.0
 pytz >= 2021.1, < 2024
 pyyaml >= 5.4.1, < 7.0.0
 readchar >= 4.0.0, < 5.0.0
 rich >= 11.0, < 14.0
 ruamel-yaml >= 0.17.0
 sqlalchemy[asyncio] >= 1.4.22, != 1.4.33, < 3.0.0
+starlette >= 0.27.0, < 0.28.0
```

Comment:
Note to self: whenever we decide to de-vendor fastapi we can replace this `startlette` pin with `fastapi` and be more-or-less back in tune with our own dependencies before we vendored it.

Reply:
Updated note to self: if we import from `starlette` then we will probably want to keep it listed here.